app/models/user.php

```diff
@@ -0,0 +1,96 @@
+<?php
+class User extends AppModel
+{
+    const AUTH_USER_KEY = 'auth_user';
+
+    public $validation = array(
+        'username'      => array(
+            'length'    => array('validate_between', 1, 16),
+            'chars'     => array('validate_username'),
+            'unique'    => array('validate_unique_name')
+        ),
+        'email'         => array(
+            'length'    => array('validate_between', 1, 30),
+            'chars'     => array('validate_email')
+        ),
+        'password'      => array(
+            'length'    => array('validate_between', 6, 20))
+    );
+
+    public function create()
+    {
+        if(!$this->validate()) {
+            throw new ValidationException('Invalid user information');
+        }
+
+        $db = DB::conn();
+        $db->insert('user', array(
+            'username'  => $this->username,
+            'email'     => $this->email,
+            'password'  => bhash($this->password))
+        );
+    }
+
+    // Checks if the username and password matches a user in the database.
+    // If there is a match, the user is logged in and the method returns true.
+    // Otherwise, returns false.
+    public static function authenticate($username, $password)
+    {
+        $db = DB::conn();
+
+        // Get the user by username
+        $row = $db->row('SELECT * FROM user WHERE username=?', array($username));
+        if(empty($row)) {
```

Comment:
please refrain from using empty($variable), use !$variable instead po :bow: 
also for conditional statements, be mindful of spacing
```php
if (condition) {
    ...
}